lib/core/src/lib/form/components/widgets/date/date.widget.ts

```diff
@@ -60,52 +61,112 @@ export class DateWidgetComponent extends WidgetComponent implements OnInit, OnDe
     maxDate: Date;
     startAt: Date;
 
-    @Input()
-    value: any = null;
+    dateInputControl: FormControl<Date>;
 
-    private onDestroy$ = new Subject<boolean>();
+    private dateChangesSubscription: Subscription;
 
-    constructor(public formService: FormService, private dateAdapter: DateAdapter<Date>) {
-        super(formService);
+    public readonly formService = inject(FormService);
+    private readonly dateAdapter = inject(DateAdapter);
+
+    ngOnInit(): void {
+        this.initFormControl();
+        this.initDateAdapter();
+        this.initDateRange();
+        this.initStartAt();
+        this.subscribeToDateChanges();
+        this.updateField();
     }
 
-    ngOnInit() {
-        if (this.field.dateDisplayFormat) {
-            const adapter = this.dateAdapter as AdfDateFnsAdapter;
-            adapter.displayFormat = this.field.dateDisplayFormat;
-        }
+    updateField(): void {
+        this.validateField();
+        this.onFieldChanged(this.field);
+    }
 
-        if (this.field) {
-            if (this.field.minValue) {
-                this.minDate = this.dateAdapter.parse(this.field.minValue, this.DATE_FORMAT);
-            }
+    private initFormControl(): void {
+        this.dateInputControl = new FormControl<Date>(
+            {
+                value: this.field?.value,
+                disabled: this.field?.readOnly || this.readOnly
+            },
+            this.isRequired() ? [Validators.required] : []
+        );
+    }
 
-            if (this.field.maxValue) {
-                this.maxDate = this.dateAdapter.parse(this.field.maxValue, this.DATE_FORMAT);
-            }
+    private subscribeToDateChanges(): void {
+        this.dateChangesSubscription = this.dateInputControl.valueChanges.subscribe((newDate: Date) => {
+            this.field.value = newDate;
+            this.updateField();
+        });
+    }
 
-            if (this.field.value) {
-                this.startAt = this.dateAdapter.parse(this.field.value, this.DATE_FORMAT);
-                this.value = this.dateAdapter.parse(this.field.value, this.DATE_FORMAT);
+    private validateField(): void {
+        if (this.dateInputControl.invalid) {
+            this.handleErrors(this.dateInputControl.errors);
+            this.field.markAsInvalid();
+        } else {
+            this.resetErrors();
+            this.field.markAsValid();
+        }
+    }
+
+    private handleErrors(errors: ValidationErrors): void {
+        const errorAttributes = new Map<string, string>();
+        switch (true) {
```

Comment:
For me it would be clearer to use if else syntax here rather than switch(true), but it's a personal preference so the change is up to you

Reply:
i think in the future we can implement general mat-errors handler component, that should handle&display errors. For now, we use the validationSummary and custom error-widget component. So once we implement the new component, we can get rid of this code from each widgets.
Issue -> https://hyland.atlassian.net/browse/AAE-24762